hermes-mock/src/main/java/pl/allegro/tech/hermes/mock/HermesMock.java

```diff
@@ -0,0 +1,147 @@
+package pl.allegro.tech.hermes.mock;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.client.ResponseDefinitionBuilder;
+import com.github.tomakehurst.wiremock.verification.LoggedRequest;
+import org.junit.rules.MethodRule;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.Statement;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Function;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.postRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlPathMatching;
+import static com.jayway.awaitility.Awaitility.await;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static java.util.stream.Collectors.toList;
+
+public class HermesMock implements MethodRule {
+
+    public enum RESPONSE_CODE {
+        CREATED(201),
+        ACCEPTED(202),
+        BAD_MESSAGE(400),
+        NOT_FOUND(404),
+        REQUEST_TIMEOUT(408),
+        INTERNAL_ERROR(500),
+        SERVICE_UNAVAILABLE(503);
+
+        private int code;
+
+        RESPONSE_CODE(int code) {
+            this.code = code;
+        }
+    }
+
+    private WireMockServer wireMockServer;
+    private ObjectMapper objectMapper = new ObjectMapper();
+    private int awaitSeconds = 5;
+
+    public HermesMock(int port) {
+        wireMockServer = new WireMockServer(port);
+    }
+
+    public void setAwaitSeconds(int awaitSeconds) {
+        this.awaitSeconds = awaitSeconds;
+    }
+
+    public void setObjectMapper(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    public void addTopic(String topicName) {
+        addTopic(topicName, RESPONSE_CODE.CREATED);
+    }
+
+    public void addTopic(String topicName, RESPONSE_CODE statusCode) {
+        addTopic(topicName, aResponse().withStatus(statusCode.code));
+    }
+
+    public void addTopic(String topicName, ResponseDefinitionBuilder responseDefinitionBuilder) {
+        wireMockServer.stubFor(get(urlPathMatching("/topics/" + topicName))
+                .willReturn(responseDefinitionBuilder
+                        .withHeader("Content-Type", "application/json")
+                        .withHeader("Hermes-Message-Id", UUID.randomUUID().toString())
+                )
+        );
+    }
+
+    public void expectSingleMessageOnTopic(String topicName) {
```

Comment:
i would add:

```
expectMessageOnTopic(String topicName, Class<T>);
```

we also need better Avro support, but it can come in separate PR